File: Iteration1-Java/src/main/java/cse3063group3/Lecturer.java

package cse3063group3;
import org.json.JSONObject;

public class Lecturer extends Staff{
    //private List<Course> coursesGiven; // will be updated in the later iterations

    // add to the uml
    public Lecturer(int id, String name, String eMail, String password){
        super(id, name, eMail, password); 
    }

    // will be omitted in this iteration
   /* public void addCourse(Course course){
        this.coursesGiven.add(course);
    }
        public List<Course> getCoursesGiven(){
        return (this.coursesGiven);
    }*/
    public  JSONObject toJSON(){return null;};
    
    //TODO: Actually implement this.
    public void mainMenu(){}
}

File: Iteration2-Python/src/Lecturer.py

from Staff import Staff

import logging
logger = logging.getLogger(__name__)    # to ensure that the logger is named after the module
class Lecturer(Staff):
    def __init__(self , id,name,eMail,password,controller_instance):
        super(Lecturer,self).__init__(id,name,eMail,password,controller_instance)
        
    def main_menu(self):
        pass
